### PR TITLE
Use single editor in ArendMessagesView (fixes #277)

### LINKED ISSUE #277: Type checking is not executed after "Replace with constructor" until I move a caret
Kapture 2021-06-20 at 15 47 54

As you can see, I apply "Replace with constructor" and the newly generated hole is not typechecked until I move a caret.

### PR DESCRIPTION
**Pull request by marat-rkh**
This pull request solves the following bug: #277. Here is why the bug happens:
1. `TypecheckerPassFactory.createPass` has a branch that doesn't create the pass, but calls `ArendMessagesView.update` (via `ArendMessagesService.update`).
2. That call can release the editor used in `ArendMessagesView`.
3. The Platform's subsystem that runs highlighting passes does not apply the results of the passes because some editor was released. This happens in `PassExecutorService#applyInformationToEditorsLater`.

To solve the issue, I rewrote `ArendMessagesView`. Now, instead of using a separate editor for each error, it uses a single one for all errors. So we never release it (as long as the current project is open).

Here is how it looks like. Before the fix:

paste-before

After the fix:

paste-after

So, basically, typechecking results are immediately shown both in Arend Errors and in the editor, you do not need to move the caret to see them.

### CODE REVIEW

**Comment by ice1000** [comment]
Wow
